fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/postgresql/PostgreSqlResourceDAO.java

```diff
@@ -143,14 +253,14 @@ public Resource  insert(Resource resource, List<ExtractedParameterValue> paramet
                 // this is just a concurrency update, so there's no need to log the SQLException here
                 throw new FHIRPersistenceVersionIdMismatchException("Encountered version id mismatch while inserting Resource");
             } else {
-                FHIRPersistenceException fx = new FHIRPersistenceException("SQLException encountered while inserting Resource.");
+                FHIRPersistenceDataAccessException fx = new FHIRPersistenceDataAccessException("SQLException encountered while inserting Resource.");
                 throw severe(logger, fx, e);
             }
         } catch(Throwable e) {
             FHIRPersistenceDataAccessException fx = new FHIRPersistenceDataAccessException("Failure inserting Resource.");
             throw severe(logger, fx, e);
         } finally {
-            this.cleanup(null, connection);
+            this.cleanup(stmt, connection);
```

Comment:
this is very strange to me

Reply:
do the cleanup same as DB2 codes, it just returns the connection to the pool